RadixWallet/Clients/AccountPortfoliosClient/AccountPortfoliosClient+Live.swift

```diff
@@ -1,61 +1,143 @@
-
 // MARK: - AccountPortfoliosClient + DependencyKey
 extension AccountPortfoliosClient: DependencyKey {
-	/// Internal state that holds all loaded portfolios.
-	actor State {
-		let portfoliosSubject: AsyncCurrentValueSubject<[AccountAddress: OnLedgerEntity.Account]> = .init([:])
+	public static let liveValue: AccountPortfoliosClient = {
+		let state = State()
 
-		func setOrUpdateAccountPortfolio(_ portfolio: OnLedgerEntity.Account) {
-			portfoliosSubject.value.updateValue(portfolio, forKey: portfolio.address)
-		}
+		@Dependency(\.onLedgerEntitiesClient) var onLedgerEntitiesClient
+		@Dependency(\.cacheClient) var cacheClient
+		@Dependency(\.tokenPricesClient) var tokenPricesClient
+		@Dependency(\.appPreferencesClient) var appPreferencesClient
+		@Dependency(\.gatewaysClient) var gatewaysClient
 
-		func setOrUpdateAccountPortfolios(_ portfolios: [OnLedgerEntity.Account]) {
-			var newValue = portfoliosSubject.value
-			for portfolio in portfolios {
-				newValue[portfolio.address] = portfolio
+		/// Update currency amount visibility based on the profile state
+		Task {
+			for try await isCurrencyAmountVisible in await appPreferencesClient.appPreferenceUpdates().map(\.display.isCurrencyAmountVisible) {
+				guard !Task.isCancelled else { return }
+				await state.setIsCurrencyAmountVisble(isCurrencyAmountVisible)
 			}
-			portfoliosSubject.value = newValue
 		}
 
-		func portfolioForAccount(_ address: AccountAddress) -> AnyAsyncSequence<OnLedgerEntity.Account> {
-			portfoliosSubject.compactMap { $0[address] }.eraseToAnyAsyncSequence()
+		/// Update used currency based on the profile state
+		Task {
+			for try await fiatCurrency in await appPreferencesClient.appPreferenceUpdates().map(\.display.fiatCurrencyPriceTarget) {
+				guard !Task.isCancelled else { return }
+				await state.setSelectedCurrency(fiatCurrency)
+			}
 		}
-	}
 
-	public static let liveValue: AccountPortfoliosClient = {
-		let state = State()
+		/// Fetches the pool and stake units details for a given account; Will update the portfolio accordingly
+		@Sendable
+		func fetchPoolAndStakeUnitsDetails(_ account: OnLedgerEntity.Account) async {
```

Comment:
No pool and stake unit details are fetched right after the initial resources are fetched, instead being fetched when users navigate to the stake or pool units tab. This is needed so we can calculate the  totalFiat worth on while being on the home screen.
This also nicely simplifies the logic in AccountDetails, when there is no need to separately trigger the load for pool and stake unit list.